### PR TITLE
Added CachingDeviceAllocator

### LINKED ISSUE #3: missing cub/util_allocator.cuh header
It seems hipCUB is missing the cub/util_allocator.cuh header which is used in PyTorch's Caffe2 backend.

### PR DESCRIPTION
**Pull request by neon60**
- Added CachingDeviceAllocator
- Fixed the version number on CUDA
Fixes #3

### CODE REVIEW

File: hipcub/include/hipcub/cub/util_allocator.hpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/******************************************************************************
+ * Copyright (c) 2011, Duane Merrill.  All rights reserved.
+ * Copyright (c) 2011-2018, NVIDIA CORPORATION.  All rights reserved.
+ *
```
Comment (saadrahim) [inline hipcub/include/hipcub/cub/util_allocator.hpp]: Modifications copyright AMD 2019 is needed here. I

File: hipcub/include/hipcub/rocprim/util_allocator.hpp [hunk 2]
```diff
@@ -0,0 +1,673 @@
+/******************************************************************************
+ * Copyright (c) 2011, Duane Merrill.  All rights reserved.
+ * Copyright (c) 2011-2018, NVIDIA CORPORATION.  All rights reserved.
+ *
```
Comment (saadrahim) [inline hipcub/include/hipcub/rocprim/util_allocator.hpp]: Modifications copyright AMD

File: test/hipcub/test_hipcub_caching_device_allocator.cpp [hunk 3]
```diff
@@ -0,0 +1,344 @@
+/******************************************************************************
+ * Copyright (c) 2011, Duane Merrill.  All rights reserved.
+ * Copyright (c) 2011-2018, NVIDIA CORPORATION.  All rights reserved.
+ *
```
Comment (saadrahim) [inline test/hipcub/test_hipcub_caching_device_allocator.cpp]: Modifications copyright AMD

File: hipcub/include/hipcub/cub/util_allocator.hpp (hunk 1 above)
Comment (neon60) [reply hipcub/include/hipcub/cub/util_allocator.hpp]: I was the reviewer on our side. Sorry for that.

Comment (neon60) [reply hipcub/include/hipcub/cub/util_allocator.hpp]: Updated.

File: hipcub/include/hipcub/rocprim/util_allocator.hpp (hunk 2 above)
Comment (neon60) [reply hipcub/include/hipcub/rocprim/util_allocator.hpp]: Updated.

File: test/hipcub/test_hipcub_caching_device_allocator.cpp (hunk 3 above)
Comment (neon60) [reply test/hipcub/test_hipcub_caching_device_allocator.cpp]: Updated.

**Pull request merged**